sklearn/datasets/openml.py

```diff
@@ -0,0 +1,200 @@
+import json
+import numbers
+import sys
+import os
+from os.path import join, exists
+from warnings import warn
+
+try:
+    # Python 2
+    from urllib2 import urlopen
+except ImportError:
+    # Python 3+
+    from urllib.request import urlopen
+
+
+from scipy.io.arff import loadarff
+import numpy as np
+
+from .base import get_data_home
+from ..externals.joblib import Memory
+from ..externals.six import StringIO
+from ..externals.six.moves.urllib.error import HTTPError
+from ..utils import Bunch
+
+_SEARCH_NAME = "https://openml.org/api/v1/json/data/list/data_name/{}/limit/1"
+_DATA_INFO = "https://openml.org/api/v1/json/data/{}"
+
+
+def _get_data_info_by_name(name, version):
+    data_found = True
+    try:
+        if version == "active":
+            json_string = urlopen(_SEARCH_NAME.format(name
+                                                      + "/status/active/"))
+        else:
+            json_string = urlopen(_SEARCH_NAME.format(name)
+                                  + "/data_version/{}".format(version))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+        else:
+            raise error
+
+    if not data_found and version != "active":
+        # might have been deactivated. will warn later
+        data_found = True
```

Comment:
this looks a little weird here, given the condition. Perhaps better in else of the try-except (which you could also do above).